### PR TITLE
Pol 251 new resource table 13

### PR DESCRIPTION
**Pull request by asalgia**
### Description
 
Remove permissions block and adding incident resource table to the policies in the https://jira.flexera.com/browse/POL-251.

### Issues Resolved

[List any existing issues this PR resolves]

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: compliance/tags/tag_checker/tag_checker.pt [hunk 1]
```diff
@@ -276,6 +276,9 @@ for (var i = 0; i < resources.length; i++) {
 end
 
 escalation "escalate_resources" do
+  automatic true
```
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply compliance/tags/tag_checker/tag_checker.pt:279]: We need the email and cfw to be seperated into two actions.  email should be automatic true.

File: compliance/unapproved_instance_types/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 1.4
+
+- remove permissions block and adding incident resource table
```
Comment (cdwilhelm) [inline compliance/unapproved_instance_types/CHANGELOG.md:5]: There were no changes to permissions block.

File: compliance/unapproved_instance_types/unapproved_instance_types.pt [hunk 3]
```diff
@@ -158,16 +158,38 @@ end
...
-# Instances with unapproved instance type details.
-| Cloud Name | Cloud Type | Instance Name | Resource UID | State | Updated On | Instance Type | Href | Tags |
-| ---------- | ---------- | ------------- | ------------ | ----- | ---------- | ------------- | ---- | ---- |
-{{ range data -}}
-| {{.cloud_name}} | {{.cloud_type}} | {{.name}} | {{.resource_uid}} | {{.state}} | {{.updated_at}} | {{.instance_type}} | {{.href}} | {{.tags}} |
-{{ end -}}
-    EOS
+    export do
+      resource_level true
+      field "id" do
```
Comment (cdwilhelm) [inline compliance/unapproved_instance_types/unapproved_instance_types.pt:163]: ID can be move to the last column

File: compliance/unapproved_instance_types/unapproved_instance_types.pt [hunk 4]
```diff
@@ -177,19 +199,16 @@ end
...
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for Approval"
-      description "Explain why you are approving the action."
-    end
-  end
+escalation "stop_instances" do
+  automatic false
+  label "Stop Instances"
+  description "Stop Instances"
```
Comment (cdwilhelm) [inline compliance/unapproved_instance_types/unapproved_instance_types.pt:211]: How about something like:  Stop selected instances with unapproved instance types.

File: cost/superseded_instance/superseded_instance.pt [hunk 5]
```diff
@@ -396,10 +396,16 @@ end
...
 escalation "report_instances" do
+  automatic true
+  label "Send Mail"
+  description "Sends incidents email"
   email $param_email
 end
 
 resolution "report_and_resize_instances_resolution" do
+  automatic false
+  label "Resize Instances"
```
Comment (cdwilhelm) [inline cost/superseded_instance/superseded_instance.pt:407]: The resolution doesn't need to be changes.  your update probably doesn't have an effect.  Also, the label really doesn't match what the resolution does.

File: cost/superseded_instance/superseded_instance.pt [hunk 6]
```diff
@@ -410,20 +416,38 @@ end
...
-{{ end -}}
-
-___
-###### Policy Applied in Account: {{ rs_project_name }} (Account ID: {{ rs_project_id }}) within Org: {{ rs_org_name }} (Org ID: {{ rs_org_id }})
-EOS
     check logic_or(lt(to_n(val(item,"run_rate")),$param_threshold), lt(to_n(val(item,"run_rate")),$param_minimum_savings_threshold))
     escalate $report_instances
+    export do
+      resource_level true
+      field "id" do
```
Comment (cdwilhelm) [inline cost/superseded_instance/superseded_instance.pt:423]: Id can be moved to the last column

File: cost/superseded_instance/superseded_instance.pt [hunk 7]
```diff
@@ -410,20 +416,38 @@ end
...
+        label "Instance ID"
+      end
+      field "resource_id" do
```
Comment (cdwilhelm) [inline cost/superseded_instance/superseded_instance.pt:426]: I don't know why you added this, isn't the same as Id?

File: cost/superseded_instance/superseded_instance.pt [hunk 8]
```diff
@@ -410,20 +416,38 @@ end
...
+        label "Resource UID"
+      end
+      field "name" do
+        label "Name"
+      end
+      field "instance_type" do
+        label "Current Instance Type"
+      end
+      field "region" do
```
Comment (cdwilhelm) [inline cost/superseded_instance/superseded_instance.pt:435]: region should be the first column and sorted.

File: cost/superseded_instance_remediation/CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 1.3
+
```
Comment (cdwilhelm) [inline cost/superseded_instance_remediation/CHANGELOG.md:4]: There were no changes to permissions block.

File: cost/superseded_instance_remediation/superseded_instance_remediation.pt [hunk 10]
```diff
@@ -152,21 +152,17 @@ end
...
-    description "Approve resize action"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for approval"
-      description "Explain why you are approving the action"
-    end
-  end
-
+  automatic false
+  label "Resize Instances"
```
Comment (cdwilhelm) [inline cost/superseded_instance_remediation/superseded_instance_remediation.pt:164]: I think resize is the wrong term, more like update. ie. Update Instances

File: cost/superseded_instance_remediation/superseded_instance_remediation.pt [hunk 11]
```diff
@@ -152,21 +152,17 @@ end
...
+  description "Resize Instances"
```
Comment (cdwilhelm) [inline cost/superseded_instance_remediation/superseded_instance_remediation.pt:165]: How about Update selected instances to new instance type

File: cost/superseded_instance_remediation/superseded_instance_remediation.pt [hunk 12]
```diff
@@ -181,19 +177,27 @@ end
...
-___
-###### Policy Applied in Account: {{ rs_project_name }} (Account ID: {{ rs_project_id }}) within Org: {{ rs_org_name }} (Org ID: {{ rs_org_id }})
-EOS
     check eq(size(data), 0)
-	escalate $email
+	  escalate $report_superseded_instance_email
     escalate $esc_resize_instances
+    export do
+      resource_level true
+      field "id" do
```
Comment (cdwilhelm) [inline cost/superseded_instance_remediation/superseded_instance_remediation.pt:185]: Id should be the last column.  datacenter should be added as the second column.  cloud should be the first column.  should be sorted by cloud, then datacenter.

File: cost/terminate_policy/CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v1.5
+
+- remove permissions block and adding incident resource table
```
Comment (cdwilhelm) [inline cost/terminate_policy/CHANGELOG.md:5]: There were no changes to permissions block.

File: cost/terminate_policy/instance_terminate.pt [hunk 14]
```diff
@@ -168,18 +164,28 @@ end
...
-| {{ .name }} | {{.terminate_date}} | {{.run_time}} | {{.locked}} |
-{{ end -}}
-EOS
     escalate $email
     escalate $report_and_terminate_instances
     check logic_not(val(item,"terminate"))
     resolve $report_resolution
+    export do
+      resource_level true
+      field "id" do
```
Comment (cdwilhelm) [inline cost/terminate_policy/instance_terminate.pt:173]: id should be the last column.  
Add cloud first column, sorted
Add datacenter as second column sorted

File: operational/vmware/instance_tag_sync/changelog.md [hunk 15]
```diff
@@ -1,8 +1,14 @@
-v1.1
-----
+# Changelog
+
+## v1.3
+
+- remove permissions block and adding incident resource table
```
Comment (cdwilhelm) [inline operational/vmware/instance_tag_sync/changelog.md:5]: There were no changes to permissions block.

File: operational/vmware/instance_tag_sync/instance_tag_sync.pt [hunk 16]
```diff
@@ -117,14 +117,21 @@ end
...
-# tags synced on {{ parameters.param_cloud_name }}
-| instanceId | name | state | tags |
-| -------- | ------ | ------- | -------- |
-{{ range data -}}
-  | {{.id}} | {{.name}} | {{.state}} | {{ range $i, $e := .tags }}{{ if $i }}, {{ end }}{{ $e }}{{ end }} |
-{{ end -}}
-EOS
+    export do
+      resource_level true
+      field "id" do
```
Comment (cdwilhelm) [inline operational/vmware/instance_tag_sync/instance_tag_sync.pt:122]: Id should be the last column
add cloud as the first column and sorted.

File: operational/vmware/instance_tag_sync/instance_tag_sync.pt [hunk 17]
```diff
@@ -135,15 +142,9 @@ end
...
-    label "Approve to Update tags"
-    description "Update tags"
-    parameter "approval_reason" do
-      type "string"
-      label "Reason for approval"
-      description "Explain why you are approving the action"
-    end
-  end
+  automatic false
+  label "Update Tags"
```
Comment (cdwilhelm) [inline operational/vmware/instance_tag_sync/instance_tag_sync.pt:146]: How about sync tags

File: compliance/tags/tag_checker/tag_checker.pt [hunk 18]
```diff
@@ -276,6 +276,9 @@ for (var i = 0; i < resources.length; i++) {
...
+  label "Send Mail and tag resources"
```
[the remark this answers is not in the document]
Comment (asalgia) [reply compliance/tags/tag_checker/tag_checker.pt:280]: Hey curt I only added the label and description to this resource from the original policy.  Is there anything other that is required?

File: compliance/tags/tag_checker/tag_checker.pt (hunk 1 above)
[the remark this answers is not in the document]
Comment (asalgia) [reply compliance/tags/tag_checker/tag_checker.pt:279]: separated out the CWF and email escalation.

File: compliance/tags/tag_checker/tag_checker.pt [hunk 19]
```diff
@@ -288,16 +291,36 @@ end
...
-| Cloud Name | Cloud Type | Name | ID | Type | State | Href | Missing Tags | Invalid Tags |
-| ---------- | ---------- | ---- | --- | ---- | ----- | ---- | ------------ | ------------ |
-{{ range data -}}
-| {{ .cloud_name }} | {{ .cloud_type }} | {{ .name }} | {{ .id }} | {{ .type }} | {{ .state }} | {{ .href }} | {{ range $i, $e := .missing_tags }}{{ if $i }}, {{ end }}{{ $e }}{{ end }} | {{ range $i, $e := .invalid_tags }}{{ if $i }}, {{ end }}{{ $e }}{{ end }}  |
-{{ end -}}
-EOS
-
+    export do
+      resource_level true
+      field "id" do
```
[the remark this answers is not in the document]
Comment (asalgia) [reply compliance/tags/tag_checker/tag_checker.pt]: moved ID to the last column.

File: compliance/unapproved_instance_types/CHANGELOG.md (hunk 2 above)
Comment (asalgia) [reply compliance/unapproved_instance_types/CHANGELOG.md:5]: Change the comment to reflect remove approval block and adding incident resource table

File: compliance/unapproved_instance_types/unapproved_instance_types.pt (hunk 3 above)
Comment (asalgia) [reply compliance/unapproved_instance_types/unapproved_instance_types.pt:163]: moved ID to the last column.

File: compliance/unapproved_instance_types/unapproved_instance_types.pt (hunk 4 above)
Comment (asalgia) [reply compliance/unapproved_instance_types/unapproved_instance_types.pt:211]: changed the description.

File: cost/superseded_instance/superseded_instance.pt (hunk 5 above)
Comment (asalgia) [reply cost/superseded_instance/superseded_instance.pt:407]: reverted resolution to the original state.

File: cost/superseded_instance/superseded_instance.pt (hunk 6 above)
Comment (asalgia) [reply cost/superseded_instance/superseded_instance.pt]: moved ID to last column.

File: cost/superseded_instance/superseded_instance.pt (hunk 7 above)
Comment (asalgia) [reply cost/superseded_instance/superseded_instance.pt:420]: I was not sure if they are the same.  My intention was to keep everything in the summary section https://github.com/flexera/policy_templates/blob/master/cost/superseded_instance/superseded_instance.pt#L416. Would you like me to remove the Resource UID field?

File: compliance/tags/tag_checker/tag_checker.pt (hunk 18 above)
[the remark this answers is not in the document]
Comment (cdwilhelm) [reply compliance/tags/tag_checker/tag_checker.pt]: You need to consider what the action does and write a label that matches.  This escalation doesn't send an email.

File: cost/superseded_instance/superseded_instance.pt (hunk 8 above)
Comment (asalgia) [reply cost/superseded_instance/superseded_instance.pt:429]: Moved the region to the top field. I not not sure how to sort the export. Can you point to some example policy which has done this on a the export?

File: cost/superseded_instance_remediation/CHANGELOG.md (hunk 9 above)
Comment (asalgia) [reply cost/superseded_instance_remediation/CHANGELOG.md:4]: Change the comment to reflect remove approval block and adding incident resource table.

File: cost/superseded_instance_remediation/superseded_instance_remediation.pt (hunk 12 above)
Comment (asalgia) [reply cost/superseded_instance_remediation/superseded_instance_remediation.pt:185]: Changed the ID to last column, cloud as first column. The policy does not get the datacenter field. Id that field that should be added to the policy. I not not sure how to sort the export. Can you point to some example policy which has done this on a the export?

File: operational/vmware/instance_tag_sync/changelog.md (hunk 15 above)
Comment (asalgia) [reply operational/vmware/instance_tag_sync/changelog.md]: made appropriate changes

File: cost/terminate_policy/CHANGELOG.md (hunk 13 above)
Comment (asalgia) [reply cost/terminate_policy/CHANGELOG.md]: made appropriate changes.

File: operational/vmware/instance_tag_sync/instance_tag_sync.pt (hunk 17 above)
Comment (asalgia) [reply operational/vmware/instance_tag_sync/instance_tag_sync.pt]: fixed the label and description

File: cost/terminate_policy/CHANGELOG.md (hunk 13 above)
Comment (asalgia) [reply cost/terminate_policy/CHANGELOG.md]: fixed .

File: compliance/tags/tag_checker/tag_checker.pt (hunk 18 above)
[the remark this answers is not in the document]
Comment (asalgia) [reply compliance/tags/tag_checker/tag_checker.pt]: separated out the escalation block of each.

File: cost/terminate_policy/instance_terminate.pt (hunk 14 above)
Comment (asalgia) [reply cost/terminate_policy/instance_terminate.pt]: added datacenter and cloud.

File: operational/vmware/instance_tag_sync/instance_tag_sync.pt (hunk 16 above)
Comment (asalgia) [reply operational/vmware/instance_tag_sync/instance_tag_sync.pt]: added cloud.

**Pull request merged**